File: 226 Invert Binary Tree/nonrec.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (!root)
            return NULL;
        queue<TreeNode*> resultseq;
        resultseq.push(root);
        while (!resultseq.empty()){
            TreeNode *helper = resultseq.front();
            resultseq.pop();
            TreeNode *tmp = helper->left;
            helper->left = helper->right;
            helper->right = tmp;
            if (helper->left)
                resultseq.push(helper->left);
            if (helper->right)
                resultseq.push(helper->right);
        }
        return root;
    }
};

File: 226 Invert Binary Tree/nonrec.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from collections import deque
class Solution:
    # @param {TreeNode} root
    # @return {TreeNode}
    def invertTree(self, root):
        if not root:
            return None
        resultseq = deque([])
        resultseq.append(root)
        while(len(resultseq)!=0):
            helper = resultseq[0]
            resultseq.popleft()
            tmp = helper.left
            helper.left = helper.right
            helper.right = tmp
            if helper.left:
                resultseq.append(helper.left)
            if helper.right:
                resultseq.append(helper.right)
        return root